crates/biome_js_parser/src/syntax/function.rs

```diff
@@ -754,34 +754,36 @@ fn is_parenthesized_arrow_function_expression_impl(
         }
         // potential start of type parameters
         T![<] => {
-            if is_nth_at_type_parameter_modifier(p, n + 1) && !JsSyntaxFeature::Jsx.is_supported(p)
-            {
-                // <const T>...
-                IsParenthesizedArrowFunctionExpression::True
-            } else if !is_nth_at_identifier(p, n + 1) {
-                // <5...
-                IsParenthesizedArrowFunctionExpression::False
-            }
             // test jsx jsx_type_arguments
             // // These may look like a valid arrows but are JSX
             // <A extends>() =</A>;
             // <A extends="B">() =</A>;
             // <A extends ok>() =</A>;
+            // <const A>() =</const>;
+            // <const A extends/>;
 
             // test tsx tsx_type_arguments
             // // These are valid type arguments
+            // <A,>() => {};
+            // <const A,>() => {};
             // <A extends B>() => {};
             // <A=string>() => {};
             // <A, B>() => {};
-            // <A extends B<C>>() => {}
-
-            // <a... JSX override
-            else if JsSyntaxFeature::Jsx.is_supported(p) {
+            // <A extends B<C>>() => {};
+
+            if JsSyntaxFeature::Jsx.is_supported(p) {
+                // Disambiguate between JSX and type parameters
+                // Type parameters of arrow functions accept only the `const` modifier.
+                let n = if p.nth(n + 1) == T![const] { n + 1 } else { n };
```

Comment:
nit: 
You can use `p.nth_at` here.

Reply:
Thanks for the suggestion. By the way, why do we have this helper? Using `nth(<n>) = <kind>` looks cleaner?

Reply:
Hm, we can discuss it.

1.
`p.nth_at(n, T![abstract]) && p.nth_at(n + 1, T![class])` 
vs 
`p.nth(n) == T![abstract] && p.nth(n + 1) == T![class]`

2.
`matches!(p.nth(n), T![abstract] | T![const]) && matches!(p.nth(n + 1), T![class] | T![const)`
 vs 
`p.nth_at_ts(n, FIRST_SET) && p.nth_at_ts(n + 1, SECOND_SET)`